src/nbl/video/CVulkanPipelineLayout.cpp

```diff
@@ -0,0 +1,18 @@
+#include "CVulkanPipelineLayout.h"
+
+#include "nbl/video/CVKLogicalDevice.h"
+
+namespace nbl::video
+{
+
+CVulkanPipelineLayout::~CVulkanPipelineLayout()
+{
+    const auto originDevice = getOriginDevice();
+    if (originDevice->getAPIType() == EAT_VULKAN)
+    {
```

Comment:
if an object has already been created, you can assume the origin device is compatible